### ISSUE TITLE
Reconfigure dev release workflow to use podman

### ISSUE THREAD

**Issue by komish** [redhat-certification/chart-verifier#391]
https://github.com/redhat-certification/chart-verifier/blob/main/.github/workflows/dev_release.yaml

In https://github.com/redhat-certification/chart-verifier/pull/386 we reconfigured our image building workflow to leverage podman. The dev release was configured to use the scripts that leveraged the docker API in order to match the main workflow. Since that's now changed, the dev release can probably be updated accordingly.

As a secondary item, the dev release `0.1.0` and the `main` image tags are built on the same triggers, but are built at different times. It may make sense to work these two things together such that they're the same underlying image.

**Comment by mgoerens** [comment]
Interestingly they differ in size (and of course also different sha):
![image](https://github.com/redhat-certification/chart-verifier/assets/1616123/23ec5d62-dd5f-4be0-84ad-81e31849e618)

**Comment by komish** [comment]
@mgoerens Indeed. For tag management, I generally like the references to be the same (i.e. shuffling tags around).

For context, the only reason we maintain a 0.1.0 release is because it allows for the openshift-tools-installer to install the latest version of chart-verifier.

I _believe_ it actually uses the GitHub release, not the container image, but the image is pushed to keep things consistent.

**Comment by mgoerens** [comment]
> In https://github.com/redhat-certification/chart-verifier/pull/386 we reconfigured our image building workflow to leverage podman. The dev release was configured to use the scripts that leveraged the docker API in order to match the main workflow. Since that's now changed, the dev release can probably be updated accordingly.

This was actually tackled in #378 where the dev_release workflow was updated to use the `build-image` make target rather than the "build-and-test" python script.

**Comment by mgoerens** [comment]
> As a secondary item, the dev release 0.1.0 and the main image tags are built on the same triggers, but are built at different times. It may make sense to work these two things together such that they're the same underlying image.

Do you suggest merging "main.yaml" and "dev_release.yaml" workflows ?

**Comment by komish** [comment]
@mgoerens thanks for correcting the record here to make sure things tie back to the right places.

> Do you suggest merging "main.yaml" and "dev_release.yaml" workflows ?

I don't like the idea of merging the two workflows, but I do want to see the images be fundamentally the same underlying digests. The easy way to do this would be to remove the image-publishing component for dev_release and just tag 0.1.0 when we build the main image.

This could also be re-architected some way to separate concerns, but I'll leave that open to whatever you like.
